hw/arm/aspeed.c

```diff
@@ -369,6 +376,23 @@ static void aspeed_machine_init(MachineState *machine)
                          amc->uart_default);
     qdev_realize(DEVICE(&bmc->soc), NULL, &error_abort);
 
+    SpiGpioState *spi_gpio = SPI_GPIO(qdev_new(TYPE_SPI_GPIO));
+    qdev_realize(DEVICE(spi_gpio), NULL, &error_fatal);
+
+    DeviceState *m25p80 = qdev_new("n25q256a");
+    qdev_realize(m25p80, BUS(spi_gpio->spi), &error_fatal);
+
+    qdev_connect_gpio_out_named(DEVICE(&bmc->soc.gpio), "sysbus-irq", AST_GPIO_IRQ_X0_NUM,
+                                qdev_get_gpio_in_named(m25p80, SSI_GPIO_CS, 0));
+    qdev_connect_gpio_out_named(DEVICE(&bmc->soc.gpio), "sysbus-irq", AST_GPIO_IRQ_X3_NUM,
+                                qdev_get_gpio_in_named(DEVICE(spi_gpio), "SPI_CLK", 0));
+    qdev_connect_gpio_out_named(DEVICE(&bmc->soc.gpio), "sysbus-irq", AST_GPIO_IRQ_X4_NUM,
+                                qdev_get_gpio_in_named(DEVICE(spi_gpio), "SPI_MOSI", 0));
+
+    object_property_set_link(OBJECT(&bmc->soc.gpio), AST_GPIO_IRQ_X5_NAME,
+                             OBJECT(qdev_get_gpio_in_named(DEVICE(spi_gpio), "SPI_MISO", 0)),
+                             &error_abort);
```

Comment:
```suggestion
    /* connect spi_gpio's output pin SPI_MISO to aspeed_gpio's GPIOX5 */
    qdev_connect_gpio_out_named(DEVICE(spi_gpio), "SPI_MISO", 0,  
        &bmc->soc.gpio.gpios[5][18]);  
```
howto get the qemu_irq reference of each GPIO pin of aspeed_gpio, we need discuss.